hoverkite-firmware/src/hoverboard.rs

```diff
@@ -26,6 +26,7 @@ use gd32f1x0_hal::{
     rcu::{Clocks, AHB, APB1, APB2},
     serial::{Config, Rx, Serial, Tx},
 };
+use hoverkite_protocol::MessageReader;
```

Comment:
The idea was for the `hoverboard` module not to have anything specific to our project, so that it could be split out to a separate board support package crate in future. So I'd rather we kept anything specific to our protocol out of it.

Reply:
This is the best way I could see to have MessageReader hold the &mut Rx (which feels like the correct ownership scheme: once you have attached the MessageReader, you don't want other things randomly stealing bytes out of the underlying stream). 

I will try to find another way (maybe it is possible to split Hoverboard into separate subsystems and avoid handing out the entire hoverboard in one go?)